bingo/bingo-elastic/python/bingo_elastic/queries.py

```diff
@@ -47,23 +47,60 @@ def compile(
 
 
 class KeywordQuery(CompilableQuery):
-    def __init__(self, value: str):
+    is_substructure: bool
+
+    def __init__(self, value: str, is_substructure: bool = False):
         self._value = value
+        self.is_substructure = is_substructure
 
     def compile(
         self,
         query: Dict,
         postprocess_actions: Optional[PostprocessType] = None,
     ) -> None:
-        bool_head = head_by_path(
-            query, ("query", "script_score", "query", "bool")
-        )
-        if not bool_head.get("must"):
-            bool_head["must"] = []
-        bool_head["must"].append(
+        if self.is_substructure:
+            bool_head = head_by_path(query, ("query", "bool"))
+        else:
+            bool_head = head_by_path(query, ("query", "script_score", "query", "bool"))
+        parent_term = "must"
+        if not bool_head.get(parent_term):
+            bool_head[parent_term] = []
+        bool_head[parent_term].append(
             {"match": {self.field: {"query": self._value, "boost": 0}}}
         )
-        default_script_score(query)
+        if not self.is_substructure:
+            default_script_score(query)
+
+
+class TermQuery(CompilableQuery):
```

Comment:
Should we add some tests for this Query? We don't have much documentation, but tests can help us to understand usage and use-cases. Also, if this PR will be merged, we need to migrate same functionality into Java wrapper and tests will help us to achieve that